### ISSUE TITLE
Question: Update partial query?

### ISSUE THREAD

**Issue by samcoenen** [apollographql/apollo-client#2747]
**Question**

Let's say I have this query which is filled with data on the client
```
query {
  conversation(conversationId: "1") {
    id
    field1
    field2
    field3
    ...
    field 45
    isClosed
  }
}
```

And I execute this mutation to update the `isClosed` field on it
```
mutation {
  updateConversation(conversationId: "1", isClosed: true) {
    id
    isClosed
  }
}
```

Currently I have to add all the same fields in the response of both the query and the mutation, otherwise Apollo will fail to update its' store with the new data (the question of whether to allow stale data or not, which has already been asked before a couple of times).

It was my understanding that GraphQL had been created as a more network-efficient solution compared to REST etc.. However, Apollo is forcing me here to write inefficient code because it requires me to query all those fields again even though I know they haven't changed. As my example shows, this can easily become a huge waste of resources in the case of many fields, especially if some of those require additional database work on the server.

In this case, I know as the developer that none of the fields have changed except `isClosed` so I don't mind having stale data in my store. If I knew there would be side-effect e.g. a `lastUpdated` field which would have changed, I could just ask for both the updated `isClosed` and the `lastUpdated` field after my mutation. To me it seems that it's the responsibility of the developer to make sure all his necessary data is fresh, Apollo shouldn't force me to re-query my entire object because it's nullifying one of the major advantages of GraphQL: the efficiency.

Is there any way I can force my wanted behaviour inside Apollo? I'd rather not have to manually write `update` blocks after every mutation to create this behaviour as it would vastly increase my lines of code and the default behaviour is just a waste of resources.

**Comment by n1ru4l** [comment]
Hey there! I tried to reproduce this issue here: https://codesandbox.io/s/6jx1mr392w but did not succeed in doing so...

To resolve your issue we need at least the version and names of the packages that you are using or an error message..

Could you please provide some more info and an example that reproduces your issue by forking my sandbox or using the react-apollo-error-template? https://github.com/apollographql/react-apollo-error-template/

**Comment by samcoenen** [comment]
Hi @n1ru4l !

Thank you for taking the time to create a sandbox! I would've done it myself but since I didn't think it was a bug but rather a design choice from Apollo I didn't bother doing so.

I couldn't believe me eyes when I saw it working in your sandbox so I'm trying to reproduce it here but awkwardly enough I can't. Always hate it when that happens. I did make quite a few changes to my code in the meantime so maybe it was related to something else.

I do however remember the error message when the issue occurred: **none**.
My props (I used the `@graphql` HoC to query the data in the first place) became undefined (not even that, it just vanished) after the mutation succeeded.

```
// props
 {
  data: {
    loading: false,
    conversation: [...], // <-- This whole field vanished entirely
    fetchMore: Function,
    ...
  }
}
```
There was no error message, no log, nothing. Using the dev tools I could see that the store was still holding the old, non-updated values. Only when I started querying all the fields, I started getting the expected results.

I'll try to reproduce this bug over the next days because no data and no log or error why is never good! If I can't reproduce it, I'll close this issue.
